strings/boyer_moore.cpp

```diff
@@ -0,0 +1,222 @@
+/**
+ * @file
+ * @brief Boyer-moore's algorithm finding all occurrences of pattern in given
+ * text.
+ * @author [Stoycho Kyosev](https://github.com/stoychoX)
+ */
+#include <cassert>  /// for assert
+#include <cstring>  /// for strlen
+#include <limits>   /// for CHAR_MAX macro
```

Comment:
`climits` should be used for this instead.
```suggestion
#include <climits>   /// for CHAR_MAX macro
```